target_redshift/db_sync.py

```diff
@@ -653,109 +646,81 @@ def load_csv(self, s3_key, count, size_bytes, compression=False, manifest=False)
+                            WHERE _sdc_deleted_at IS NOT NULL
+                        """
+                        )
 
-                        # insert into history where sys_end_time is not null
-                        update_sql = """
-                        INSERT INTO {} ({})
-                        SELECT {}
-                        FROM {} s
+                        run(
+                            f"""
+                        INSERT INTO {history_table} ({columns})
+                        SELECT {stage_columns}
+                        FROM {target_table} s
                         WHERE _sys_end_time IS NOT NULL
-                        """.format(
-                            history_table,
-                            ", ".join([c["name"] for c in columns_with_trans]),
-                            ", ".join(
-                                ["s.{}".format(c["name"]) for c in columns_with_trans]
-                            ),
-                            target_table,
+                        """
                         )
-                        self.logger.info("Running query: {}".format(update_sql))
-                        cur.execute(update_sql)
-
-                        # delete anything that has a _sys_end_time set on it
-                        update_sql = """
-                        DELETE FROM {} WHERE _sys_end_time IS NOT NULL
-                        """.format(
-                            target_table
+
+                        run(
+                            f"""
+                        DELETE FROM {target_table} WHERE _sys_end_time IS NOT NULL
+                        """
                         )
 
-                        self.logger.info("Running query: {}".format(update_sql))
-                        cur.execute(update_sql)
                         updates = cur.rowcount
 
                     # Step 5/a/2: Insert new records
-                    insert_sql = """INSERT INTO {} ({})
-                        SELECT {}
-                        FROM {} s
-                    """.format(
-                        target_table,
-                        ", ".join([c["name"] for c in columns_with_trans]),
-                        ", ".join(
-                            ["s.{}".format(c["name"]) for c in columns_with_trans]
-                        ),
-                        stage_table,
+                    run(
+                        f"""INSERT INTO {target_table} ({columns})
+                        SELECT {stage_columns}
+                        FROM {stage_table} s
+                        WHERE _sdc_deleted_at IS NOT NULL
+                    """
                     )
-                    self.logger.info("Running query: {}".format(insert_sql))
-                    try:
-                        cur.execute(insert_sql)
-                    except Exception as e:
-                        self.logger.info(
-                            "FAILED AT RUNNING QUERY: {}".format(insert_sql)
```

Comment:
a lot of this was swallowing errors which I really don't like doing. So this unravels that a bit.